racetrack_client/racetrack_client/main.py

```diff
@@ -63,9 +64,23 @@ def _deploy(
     extra_vars: Optional[List[str]] = typer.Option(None, '-e', '--extra-vars', help='key=value pairs overriding manifest values'),
 ):
     """Send request deploying a Job to the Racetrack cluster"""
+    # Catch if something was piped to `deploy`. Works best when the build_context is git
+    # Since most of send_deploy_request assumes a directory and a manifest file, we create one from the piped data
+    piped_data = ""
+    file_path = ""
+    if not sys.stdin.isatty() and (piped_data := sys.stdin.read()):
+        current_workdir = os.getcwd()
+        with tempfile.NamedTemporaryFile(mode='w+', dir=current_workdir, delete=False) as tmp_file:
+            tmp_file.write(piped_data)
+            file_path = tmp_file.name
+        workdir = file_path
+
```

Comment:
Nitpick: I think it all could be moved to racetrack_client/client/deploy.py, keeping this logic away from main.py

Reply:
I think some of it _has_ to be in main to capture piped input. I'll most of the logic to deploy.py, as I agree that main.py should be clean.